config/provider/upgrade/config.go

```diff
@@ -0,0 +1,11 @@
+package providerupgradeconfig
+
+type Configurations struct {
+	Providers []ProviderConfiguration
+}
+
+type ProviderConfiguration struct {
```

Comment:
nitpick
```suggestion
type Provider struct {
```
Maybe also consider `ProviderTest` or `ProviderConformance`. Just thinking about using similar tests for `Configuration` packages and it being called `ConfigurationConfiguration`, which would be a little confusing :)

Feel free to push back on any naming suggestions / offer up your own thoughts!